src/msg.c

```diff
@@ -394,50 +378,14 @@ static bool property_changed(const char *name,
 	struct l_queue *list;
 	struct session *session;
 	struct knot_device *device;
-	struct mydevice *mydevice;
 	char id[KNOT_ID_LEN];
 
 	/* FIXME: manage link overload or not connected */
 	session = l_queue_find(session_list, session_node_fd_cmp, user_data);
 	if (!session)
 		return false;
 
-	/* FIXME: Memory leak & detect if schema has changed */
-	if (strcmp("schema", name) == 0) {
-		if (session->schema && strcmp(session->schema, value) == 0)
-			goto done;
-
-		/* Track to detect if update is required */
-		list = parser_schema_to_list(value);
-		if (list == NULL) {
-			hal_log_error("[session %p] schema: parse error!",
-				      session);
-			goto done;
-		}
-
-		if (l_queue_find(list, schema_find_invalid, NULL)) {
-			hal_log_error("[session %p] schema: consistency error!",
-				      session);
-			goto done;
-		}
-
-		/* TODO: Verify if schema received is valid */
-		l_queue_destroy(session->schema_list, l_free);
-		session->schema_list = list;
-		l_free(session->schema);
-		session->schema = l_strdup(value);
-
-		snprintf(id, sizeof(id), "%016"PRIx64, session->id);
-
-		mydevice = l_queue_find(device_id_list, device_id_cmp, id);
-		if (!mydevice)
-			goto done;
-
-		device = device_get(id);
-		if (device)
-			device_set_registered(device, true);
```

Comment:
This must be done in the new function `handle_schema_updated`

Reply:
Done.